### PR TITLE
Increase QUIC code coverage

### PR DESCRIPTION
**Pull request by RyanTheOptimist**
Signed-off-by: Ryan Hamilton <rch@google.com>

Fixes: #23128

Risk Level: Low
Testing: Lots more tests :)
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

File: test/common/quic/quic_filter_manager_connection_impl_test.cc [hunk 1]
```diff
@@ -70,5 +70,69 @@ TEST_F(QuicFilterManagerConnectionImplTest, UpdateBytesBuffered) {
...
+  EXPECT_ENVOY_BUG(impl_.readDisable(true), "Unexpected call to readDisable");
+}
+
+TEST_F(QuicFilterManagerConnectionImplTest, DetectEarlyCloseWhenReadDisabled) {
+  EXPECT_ENVOY_BUG(impl_.detectEarlyCloseWhenReadDisabled(true),
+                   "Unexpected call to detectEarlyCloseWhenReadDisabled");
+}
+
+TEST_F(QuicFilterManagerConnectionImplTest, ConnectionInfoSetter) {
+  /*
```
Comment (alyssawilk) [inline test/common/quic/quic_filter_manager_connection_impl_test.cc:93]: should we remove this one?

**Review by alyssawilk — COMMENTED** [review]
Awesome!  Thanks for bumping this back up
/wait

File: test/common/quic/quic_filter_manager_connection_impl_test.cc (hunk 1 above)
Comment (RyanTheOptimist) [reply test/common/quic/quic_filter_manager_connection_impl_test.cc]: Whoops, done.

**Review by alyssawilk — APPROVED** [review]

**Comment by alyssawilk** [comment]
didn't this have an associated issue we can mark as being fixed?

**Comment by RyanTheOptimist** [comment]
> didn't this have an associated issue we can mark as being fixed?

Indeed. Updated!

**Pull request merged**